GMGoogleDrive/Public/Set-GDriveItemContent.ps1

```diff
@@ -282,7 +282,12 @@ function Set-GDriveItemContent {
                         ))
                     {
                         Write-Verbose 'Single request upload'
-                        $WebRequestParams.Body = $RawContent
+                        [byte[]]$buffer = New-Object byte[] $stream.Length
```

Comment:
May you explain the essence of the problem?
because in fact you are translating here [byte[]] to memorystream and back to [byte[]]

Reply:
I'm not that familiar with the memory stream.

I had an issue uploading files using the InFile Parameter, when the file size was smaller than the ChunkSize (4MB).
In the code there were to different methods "Single request upload" and "Multiple request upload". In the single request upload your version used $RawContent that is not used if you are running the command with the parameter InFile. As the multiple request upload was working I just copied some code from multi request upload to the single request upload.
If you have a better solution, please let me know.

Reply:
hmm, apparently this error has been going on since the very first release
It's strange that no one has come across it before.
I can’t say that my solution will be better, but it would probably be more correct to at least not copy blocks for nothing in the case of `-RawContent` usage
i.e. something like this

``` powershell
if ($PSCmdlet.ParameterSetName -in 'fileName','fileMeta') {
  $RawContent = New-Object byte[] $stream.Length
  $len = $stream.Read($RawContent, 0, $stream.Length);
  if ($len -ne $stream.Length) {
      throw "Stream read error: Readed $len bytes instead of $($stream.Length)"
  }
}
$WebRequestParams.Body = $RawContent
```

Reply:
bw, MemoryStream it just a stream interface around byte array
internally (in pseudocode)
```$stream = New-Object System.IO.MemoryStream $RawContent``` is just
```
public MemoryStream($Buffer, $writable) {
   this_.buffer = $Buffer
    ...
}
```

Reply:
Thanks for your explanation.
I would try to keep "Single request upload" and "Multiple request upload" as similar as possible. But I would leave the decision up to you.

Reply:
ok, then I'll just end the conversation
and when you tell me that you are ready, I will merge your PR
later when I add tests for this bug maybe I'll rewrite this section

for other cmdlets that might need the `-Parameter`, I'll add it as well